services/oauth/v2/oauth.go

```diff
@@ -0,0 +1,368 @@
+	authStatusToggleStats.isCallToCpApi = true
+	authStatusToggleStats.SendCountStat()
+
+	cpiCallStartTime := time.Now()
+	statusCode, respBody = authErrHandler.CpConn.CpApiCall(authStatusInactiveCpReq)
+	authStatusToggleStats.statName = GetOAuthActionStatName("request_latency")
+	defer authStatusToggleStats.SendTimerStats(cpiCallStartTime)
+	authErrHandler.Logger.Errorf(`Response from CP(stCd: %v) for auth status inactive req: %v`, statusCode, respBody)
+
+	var authStatusToggleRes *AuthStatusToggleResponse
+	unmarshalErr := json.Unmarshal([]byte(respBody), &authStatusToggleRes)
+	if router_utils.IsNotEmptyString(respBody) && (unmarshalErr != nil || !router_utils.IsNotEmptyString(authStatusToggleRes.Message) || statusCode != http.StatusOK) {
+		var msg string
+		if unmarshalErr != nil {
+			msg = unmarshalErr.Error()
+		} else {
+			msg = fmt.Sprintf("Could not update authStatus to inactive for destination: %v", authStatusToggleRes.Message)
+		}
+		authStatusToggleStats.statName = GetOAuthActionStatName("failure")
+		authStatusToggleStats.errorMessage = msg
+		authStatusToggleStats.SendCountStat()
+		return http.StatusBadRequest, ErrPermissionOrTokenRevoked.Error()
+	}
+
+	authErrHandler.Logger.Errorf("[%s request] :: (Write) auth status inactive Response received : %s\n", loggerNm, respBody)
+	authStatusToggleStats.statName = GetOAuthActionStatName("success")
+	authStatusToggleStats.errorMessage = ""
+	authStatusToggleStats.SendCountStat()
+
+	return http.StatusBadRequest, ErrPermissionOrTokenRevoked.Error()
+}
+
+func (authErrHandler *OAuthHandler) GetRefreshTokenErrResp(response string, accountSecret *AccountSecret) (errorType, message string) {
+	if gjson.Get(response, ErrorType).String() != "" {
+		errorType = gjson.Get(response, ErrorType).String()
+		message = gjson.Get(response, "message").String()
+	} else if err := json.Unmarshal([]byte(response), &accountSecret); err != nil {
+		// Some problem with AccountSecret unmarshalling
+		authErrHandler.Logger.Debugf("Failed with error response: %v\n", err)
+		message = fmt.Sprintf("Unmarshal of response unsuccessful: %v", response)
+		errorType = "unmarshallableResponse"
+	} else if gjson.Get(response, "body.code").String() == REF_TOKEN_INVALID_GRANT {
+		// User (or) AccessToken (or) RefreshToken has been revoked
+		bodyMsg := gjson.Get(response, "body.message").String()
+		if bodyMsg == "" {
+			// Default message
+			authErrHandler.Logger.Debugf("Failed with error response: %v\n", response)
+			message = ErrPermissionOrTokenRevoked.Error()
+		} else {
+			message = bodyMsg
+		}
+		errorType = REF_TOKEN_INVALID_GRANT
+	}
+	return errorType, message
+}
+
+// This method hits the Control Plane to get the account information
+// As well update the account information into the destAuthInfoMap(which acts as an in-memory cache)
+func (authErrHandler *OAuthHandler) fetchAccountInfoFromCp(refTokenParams *RefreshTokenParams, refTokenBody RefreshTokenBodyParams,
```

Comment:
We are using two different receivers name here: `oauthHandler` and `authErrHandler`